### DISCUSSION TITLE
On the conda/miniconda install method

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#409] [Ideas]
As I struggle with the MacOS issues people are having, I was looking into the conda/miniconda install instructions, and I searching a bit around, thinking if it would be possible so simplify things further, I saw this: https://anaconda.org/conda-forge/openjdk (is it JDK-21?)

Is JDK-21 compatible with py5, and would it be helpful to make the conda install recipe download it?

Would it be possible to make a .sh script maybe that went through all the Mac installation steps, including even Thonny and thonny-py5mode plugin?

**Comment by hx2A** [comment]
> Is JDK-21 compatible with py5, and would it be helpful to make the conda install recipe download it?

Yes, it should work fine! any Java version 17 or after should be ok.

My past experiences with installing Java through anaconda were bad, but perhaps this could work. If you create an anaconda environment file, make sure you pin the version so that specific package is installed.

What are the current MacOS installation problems?

> Would it be possible to make a .sh script maybe that went through all the Mac installation steps, including even Thonny and thonny-py5mode plugin?

Maybe! It is worth a try.

**Reply by villares** [reply]
This is a long-shot, and I guess I don't have the energy to try it right now... but it is worth knowing JDK-21 would be usable in case I get someone to try this avenue. Maybe the already proven JDK-install library can be installed and executed as well.

The MacOS issues have to do with the Thonny plug-in https://github.com/tabreturn/thonny-py5mode/issues/52 they are hard to pinpoint because I only get glimpses of them from incomplete reports on the Domestika forum... I'm pestering my Mac-wielding friends to try to reproduce them, but still no luck.

**Reply by hx2A** [reply]
I feel like we always mange to fix the MacOS issues but I agree the setup should be stress-free, and right now it is not.

**Reply by Maltiec** [reply]
the same thing with Win installation, especially if some Java (not 17) is already installed. May be there is some way to force conda making $JAVA_HOME variable automatically after Java installation?

**Reply by hx2A** [reply]
@Maltiec, good to hear from you. There probably is a file in the anaconda `envs` directory where environment variables are set, but I poked around and couldn't find it. But I don't think such a method would be easier than putting it in a `.bashrc` file or setting an environment variable through Windows.
